drivers: clock_control: stm32h5: Add MCO sources and dividers

Dalachowsky opened the pull request:
STM32H5 series lacked support for MCO configuration. Added SOC_SERIES_STM32H5X to approperiate kconfig MCO source configurations. Added new MCO sources from H5 series and updated the clock_stm32_ll_h5.c with MCO configuration.

Dalachowsky commented on drivers/clock_control/Kconfig.stm32:
```diff
@@ -150,11 +158,13 @@ config CLOCK_STM32_MCO1_DIV
 	default 1
 	range 1   5 if SOC_SERIES_STM32F4X || SOC_SERIES_STM32F7X
 	range 1   16 if SOC_SERIES_STM32L4X
+	range 1   15 if SOC_SERIES_STM32H5X
```
That's right I'll fix it in a moment. Should i just specify the range like:
```
range 1 5  if !SOC_SERIES_STM32H5X
range 1 15 if SOC_SERIES_STM32H5X
```

FRASTM replied:
yes, or like  this is done for _LL_RCC_MCO1_DIV_ :

```
	range 1   5 if SOC_SERIES_STM32F4X || SOC_SERIES_STM32F7X
	range 1   15 if SOC_SERIES_STM32H5X
```

- and also  change the description is `help` section too,

(Moreover the range 1..5 also exists for SOC_SERIES_STM32F2X and range 1..15 also exists for SOC_SERIES_STM32H7X, SOC_SERIES_STM32MP1X but that's not the purpose here)

faloj commented:
This PR will conflict with #60448 wich add MCO support for the H7 family but also do some code factorization to avoid  stm32_clock_control_mco_init duplication. Maybe wait for it to be merged before merging this one to benefit from the code factorization ?

Dalachowsky commented:
@faloj Thanks for mentioning your PR.
I was also considering extracting mco to separate common file so I'll wait until yours is accepted.

faloj commented:
PR #60448 has been merged ! You can integrate the changements into your work ;)

Dalachowsky committed:
drivers: clock_control: stm32h5: Add MCO configuration

STM32H5 series lacked support for MCO configuration.
Added SOC_SERIES_STM32H5X to approperiate kconfig MCO source
configurations. Added new MCO sources from H5 series and
updated the clock_stm32_ll_h5.c with MCO configuration.

Signed-off-by: Kacper Dalach <dalachowsky@gmail.com>

erwango commented on drivers/clock_control/Kconfig.stm32:
```diff
@@ -66,7 +66,8 @@ config CLOCK_STM32_MCO1_SRC_LSE
 	           SOC_SERIES_STM32F7X || \
 	           SOC_SERIES_STM32L4X || \
-	           SOC_SERIES_STM32H7X
+			   SOC_SERIES_STM32H7X || \
+			   SOC_SERIES_STM32H5X
```
Please fix indentation.
Apply everywhere

Dalachowsky replied:
I have fixed the indentation and applied suggestions of @faloj
Sorry for making a bit of mess in this PR

faloj commented on drivers/clock_control/Kconfig.stm32:
```diff
@@ -199,22 +206,37 @@ config CLOCK_STM32_MCO2_SRC_HSE
 	help
 	  Use CSI as source of MCO2
 
+config CLOCK_STM32_MCO2_SRC_PLL1P
```
This clock source already exist and is named "CLOCK_STM32_MCO2_SRC_PLLPCLK"

faloj replied:
This should be removed now that you added "CLOCK_STM32_MCO2_SRC_PLLPCLK" for SOC_SERIES_STM32H5 ;)

faloj commented on drivers/clock_control/Kconfig.stm32:
```diff
@@ -199,22 +206,37 @@ config CLOCK_STM32_MCO2_SRC_HSE
+	help
+	  Use PLL1P as source of MCO2
+
+config CLOCK_STM32_MCO2_SRC_PLL2P
```
Same here with "CLOCK_STM32_MCO2_SRC_PLL2PCLK"
